src/game_functions/board_functions.h

```diff
@@ -0,0 +1,100 @@
+#ifndef BOARD_FUNCTIONS_H
+#define BOARD_FUNCTIONS_H
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "game_structs.h"
+
+
+/* struct used to return position given an index. */
+struct positionFromIndex {
+    int x;
+    int y;
+};
+
+
+/**
+ * A utility function to calculate position given an index.
+ * @index the index in the 1D array.
+ * @width the width of the grid.
```

Comment:
Does the ``width of the grid`` refer to the width of the board (used in other functions)? If it does then please specify in the comments that this is the case. If it does not refer to that, then maybe add more clarity to the meaning of the comments.

Reply:
It can be the width of any 2D grid array. This is a utility function that applies to 2D array grids in general. @Porkius